pandas/tools/pivot.py

```diff
@@ -136,8 +136,14 @@ def pivot_table(data, values=None, index=None, columns=None, aggfunc='mean',
 
     table = agged
     if table.index.nlevels > 1:
-        to_unstack = [agged.index.names[i] or i
-                      for i in range(len(index), len(keys))]
+        index_names = agged.index.names[:len(index)]
```

Comment:
But when calling `unstack(level=1)` in the example above, it is ambiguous whether to unstack  by level name or the level position since both are valid level identifiers in `unstack()` (but will to prioritize level name).